### PR TITLE
fix(search-client): reduce gap between input box and recent searches 

### PR DESCRIPTION
**Pull request by barleendhaliwal**
reduce gap between input box and recent searches if no result found & do not open drop down in case no recent searches

## Description

Reduce gap between input box and recent searches if no result found & do not open drop down in case no recent searches

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/search/projects/search-lib/src/lib/search/search.component.html [hunk 1]
```diff
@@ -39,63 +39,62 @@
 </div>
 
 <div class="search-container">
-  <div *ngIf="suggestionsDisplay" class="search-popup">
+  <div *ngIf="suggestionsDisplay && recentSearches.length" class="search-popup">
```
Comment (akshatdubeysf) [inline packages/search/projects/search-lib/src/lib/search/search.component.html:42]: suggestions would not be visible if there are no recentSearches?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: packages/search/projects/search-lib/src/lib/search/search.component.html (hunk 1 above)
Comment (barleendhaliwal) [reply packages/search/projects/search-lib/src/lib/search/search.component.html]: fixed

File: packages/search/projects/search-lib/src/lib/search/search.component.ts [hunk 2]
```diff
@@ -351,4 +333,35 @@ export class SearchComponent<T extends IReturnType>
       return [category];
     }
   }
+  getModelFromModelName(name: string) {
+    let model: IModel = {name: ALL_LABEL, displayName: ALL_LABEL};
+    this.config.models.every(item => {
```
Comment (akshatdubeysf) [inline packages/search/projects/search-lib/src/lib/search/search.component.ts:338]: use find ?

File: packages/search/projects/search-lib/src/lib/search/search.component.ts [hunk 3]
```diff
@@ -351,4 +333,35 @@ export class SearchComponent<T extends IReturnType>
...
+      return true;
+    });
+    return model;
+  }
+  getModelsWithSuggestions() {
+    const modelsWithSuggestions: {model: IModel; items: T[]}[] = [];
+    const sources: string[] = [];
+    this.suggestions.forEach(suggestion => {
+      if (sources.indexOf(suggestion['source']) >= 0) {
+        modelsWithSuggestions.every(modelWithSuggestions => {
```
Comment (akshatdubeysf) [inline packages/search/projects/search-lib/src/lib/search/search.component.ts:352]: Why use every instead of forEach?

Comment (barleendhaliwal) [reply packages/search/projects/search-lib/src/lib/search/search.component.ts:352]: every allows me to break from loop and unnecessary iteration can be avoided

File: packages/search/projects/search-lib/src/lib/search/search.component.ts (hunk 2 above)
Comment (barleendhaliwal) [reply packages/search/projects/search-lib/src/lib/search/search.component.ts]: fixed

File: packages/search/projects/search-lib/src/lib/search/search.component.ts (hunk 3 above)
Comment (akshatdubeysf) [reply packages/search/projects/search-lib/src/lib/search/search.component.ts:344]: ok

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**